src/outfit.cpp

```diff
@@ -40,31 +40,28 @@ bool Outfits::loadFromXml()
 			continue;
 		}
 
-		outfits[type].emplace_back(
-		    outfitNode.attribute("name").as_string(), pugi::cast<uint16_t>(lookTypeAttribute.value()),
-		    outfitNode.attribute("premium").as_bool(), outfitNode.attribute("unlocked").as_bool(true));
+		std::string name = outfitNode.attribute("name").as_string();
+		uint16_t lookType = pugi::cast<uint16_t>(lookTypeAttribute.value());
+		bool premium = outfitNode.attribute("premium").as_bool();
+		bool unlocked = outfitNode.attribute("unlocked").as_bool(true);
+
+		Outfit outfit = Outfit(name, lookType, premium, unlocked);
+		outfits[type].emplace_back(outfit);
+		mappedOutfits[lookType] = &outfit;
```

Comment:
This will store pointers to the stack, which will become invalid as soon as the block ends.